check.ml

```diff
@@ -99,19 +79,15 @@ let well_formed prog =
         | Simple e -> check_simple_expr e
         | Op (_op, xs) ->
           List.iter check_simple_expr xs in
-      let check_arg = function
-        | Arg_by_val e -> check_expr e
-        | Arg_by_ref x -> () in
+      let check_arg = check_expr in
```

Comment:
In the type I kept `argument` a separate type, currently a synonym of `expression`. I kept these aliases as well, in case we separate the two types again.